apps/examples/testcase/le_tc/kernel/itc_libc_pthread.c

```diff
@@ -0,0 +1,322 @@
+  * Private Functions
+ ****************************************************************************/
+
+/**
+* @fn                   :itc_libc_pthread_pthread_attr_init_destroy_p_multitime
+* @brief                :pthread_attr_init and pthread_attr_destroy call multiple time
+* @Scenario             :If pthread_attr is NULL, ENOMEM is returned.
+*                        Else, it return OK and pthread_attr is set to default value
+* @API'scovered         :pthread_attr_init
+* @Preconditions        :none
+* @Postconditions       :none
+* @return               :void
+*/
+static void itc_libc_pthread_pthread_attr_init_destroy_p_multitime(void)
+{
+	pthread_attr_t attr;
+	pthread_attr_t destroyed_attr;
+	int ret_chk;
+	int i;
+
+	memset(&destroyed_attr, 0, sizeof(pthread_attr_t));
+
+	for (i = 0; i < LOOP_SIZE; i++) {
+		ret_chk = pthread_attr_init(&attr);
+		TC_ASSERT_EQ("pthread_attr_init", ret_chk, OK);
+		TC_ASSERT_EQ("pthread_attr_init", attr.stacksize, PTHREAD_STACK_DEFAULT);
+		TC_ASSERT_EQ("pthread_attr_init", attr.priority, PTHREAD_DEFAULT_PRIORITY);
+		TC_ASSERT_EQ("pthread_attr_init", attr.policy, PTHREAD_DEFAULT_POLICY);
+		TC_ASSERT_EQ("pthread_attr_init", attr.inheritsched, PTHREAD_EXPLICIT_SCHED);
+
+		ret_chk = pthread_attr_destroy(&attr);
+		TC_ASSERT_EQ("pthread_attr_destroy", ret_chk, OK);
+		TC_ASSERT_EQ("pthread_attr_destroy", memcmp(&attr, &destroyed_attr, sizeof(pthread_attr_t)), 0);
+	}
+	TC_SUCCESS_RESULT();
+	return;
+}
+
+/**
+* @fn                   :itc_libc_pthread_pthread_rwlock_init_destroy_p_multitime
+* @brief                :This tc tests pthread_rwlock_init and pthread_rwlock_destroy
+* @API's covered        :pthread_rwlock_init,pthread_rwlock_destroy
+* @Preconditions        :none
+* @Postconditions       :none
+* @return               :void
+*/
+
+static void itc_libc_pthread_pthread_rwlock_init_destroy_p_multitime(void)
+{
+	int status;
+	int i;
+	pthread_rwlock_t rw_lock;
+	rw_lock.num_readers = 10;
+	rw_lock.num_writers = 10;
+	rw_lock.write_in_progress = true;
+
+
+	for (i = 0; i < LOOP_SIZE; i++) {
+		status = pthread_rwlock_init(&rw_lock, NULL);
+		TC_ASSERT_EQ("pthread_rwlock_init", status, OK);
+		TC_ASSERT_EQ("pthread_rwlock_init", rw_lock.num_readers, 0);
+		TC_ASSERT_EQ("pthread_rwlock_init", rw_lock.num_writers, 0);
+		TC_ASSERT_EQ("pthread_rwlock_init", rw_lock.write_in_progress, false);
+
+		status = pthread_rwlock_destroy(&rw_lock);
+		TC_ASSERT_EQ("pthread_rwlock_destroy", status, OK);
+	}
+
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @fn                   :itc_libc_pthread_pthread_rwlock_init_n_attr_set
+* @brief                :This tc tests pthread_rwlock_init and pthread_rwlock_destroy
+* @API's covered        :pthread_rwlock_init,pthread_rwlock_destroy
+* @Preconditions        :none
+* @Postconditions       :none
+* @return               :void
+*/
+
+static void itc_libc_pthread_pthread_rwlock_init_destroy_n_attr_set(void)
```

Comment:
ITC name and function description name are not same

Reply:
ITC name has "destroy" used, but actually in testcase destroy API is not used. Please correct it. Use meaningful ITC name